packages/client/components/AddMissingJiraFieldModal.tsx

```diff
@@ -0,0 +1,164 @@
+import Icon from './Icon'
+import PrimaryButton from './PrimaryButton'
+import SecondaryButton from './SecondaryButton'
+import useAtmosphere from '../hooks/useAtmosphere'
+import useMutationProps from '../hooks/useMutationProps'
+import PokerSetFinalScoreMutation from '../mutations/PokerSetFinalScoreMutation'
+import AddMissingJiraFieldMutation from '../mutations/AddMissingJiraFieldMutation'
+import AddAtlassianAuthMutation from '../mutations/AddAtlassianAuthMutation'
+import AtlassianClientManager from '../utils/AtlassianClientManager'
+import AtlassianManager from '../utils/AtlassianManager'
+
+interface Props {
+  closePortal: () => void
+  meetingId: string
+  stageId: string
+  teamId: string
+  finalScore: string
+}
+
+const StyledDialogContainer = styled(DialogContainer)({
+  width: 480
+})
+
+const ButtonGroup = styled('div')({
+  marginTop: '24px',
+  display: 'flex',
+  justifyContent: 'flex-end'
+})
+
+const StyledTip = styled('p')({
+  fontSize: 13,
+  lineHeight: '16px',
+  margin: 0,
+  padding: '0 0 16px'
+})
+
+const ErrorWrapper = styled('div')({
+  alignItems: 'center',
+  color: PALETTE.TOMATO_500,
+  display: 'flex',
+  padding: 8,
+  marginTop: 8,
+  width: '100%'
+})
+
+const StyledIcon = styled(Icon)({
+  color: PALETTE.TOMATO_500,
+  fontSize: ICON_SIZE.MD24,
+  marginRight: 8
+})
+
+const StyledPrimaryButton = styled(PrimaryButton)({
+  marginLeft: 16
+})
+
+const Label = styled('div')({
+  fontSize: 14,
+  fontWeight: 600
+})
+
+const useAddMissingJiraField = ({closePortal, teamId, finalScore, meetingId, stageId}: Props) => {
+  const atmosphere = useAtmosphere()
+  const mutationProps = useMutationProps()
+
+  /**
+   * Finally, when new field is added, update the story points
+   */
+  const onAddMissingJiraFieldCompleted = useCallback(() => {
+    PokerSetFinalScoreMutation(atmosphere, {finalScore, meetingId, stageId}, mutationProps)
+  }, [atmosphere, finalScore, meetingId, stageId, mutationProps])
+
+  /**
+   * When Atlassian auth is successfully updated, add a new field to a Jira configuration on behalf of the user
+   */
+  const onAddAtlassianAuthCompleted = useCallback(() => {
+    AddMissingJiraFieldMutation(
+      atmosphere,
+      {meetingId, stageId},
+      {onError: mutationProps.onError, onCompleted: onAddMissingJiraFieldCompleted}
+    )
+  }, [atmosphere, meetingId, stageId, mutationProps, onAddMissingJiraFieldCompleted])
+
+  /**
+   * Executed when user successfully finishes the OAuth flow of Adding a new permission [jira:manage-project]
+   */
+  const onAtlassianOAuthCompleted = useCallback(
+    (code) => {
+      mutationProps.submitMutation()
+
+      AddAtlassianAuthMutation(
+        atmosphere,
+        {code, teamId},
+        {onError: mutationProps.onError, onCompleted: onAddAtlassianAuthCompleted}
+      )
+    },
+    [mutationProps.submitMutation, teamId, onAddAtlassianAuthCompleted]
+  )
+
+  const askForManageProjectPermission = useCallback(() => {
```

Comment:
I tend to use `useCallback` because if the variable is recreated on every render and gets passed somewhere in a dependency array it's causing a hook to fire every single time. I agree though, it looks way better this way and it can be safely done that way. TIL about V8 and the way it behaves with closures. Interesting! 

Fixed. 🙏

Reply:
true true! if performance ever becomes an issue we could write `useEventCallback(onClick)`, which will guarantee a persisted object and only use 1 hook!